.github/workflows/copilot_deploy.yml

```diff
@@ -45,72 +45,76 @@ jobs:
 
   pre_deploy_tests:
     secrets:
-      E2E_PAT: ${{secrets.E2E_PAT}}
-      GOV_NOTIFY_API_KEY: ${{secrets.GOV_NOTIFY_API_KEY}}
+      E2E_PAT: ${{ secrets.E2E_PAT }}
+      GOV_NOTIFY_API_KEY: ${{ secrets.GOV_NOTIFY_API_KEY }}
     uses: communitiesuk/funding-service-design-workflows/.github/workflows/pre-deploy.yml@main
     with:
       postgres_unit_testing: false
 
-  dev_copilot_deploy:
-    if: inputs.environment == 'dev' || inputs.environment == ''
-    needs: [pre_deploy_tests, paketo_build]
-    concurrency: deploy-dev
-    permissions:
-      id-token: write # This is required for requesting the JWT
-      contents: read  # This is required for actions/checkout
-    secrets:
-      AWS_ACCOUNT: ${{ secrets.AWS_ACCOUNT }}
-    uses: ./.github/workflows/environment.yml
-    with:
-      workspace: 'dev'
+  copilot_environments_workflow_setup:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.copilot_env_matrix.outputs.env_list }}
+    steps:
+      - id: copilot_env_matrix
+        run: |
+          if [ "${{ inputs.environment }}" != '' ]; then
+            echo "env_list=[\"${{ inputs.environment }}\"]" >> "$GITHUB_OUTPUT"
+          elif [ "$(( github.ref }}" == 'refs/heads/main' ];
+            echo "env_list=[\"dev\", \"test\", \"uat\", \"production\"]" >> $GITHUB_OUTPUT
+          else
+            echo "env_list=[\"dev\", \"test\", \"uat\", \"production\"]" >> $GITHUB_OUTPUT
```

Comment:
I meant we only want dev here, don't we?  This is releasing onto a branch so shouldn't go further.

Reply:
Ah, I see what you mean now! Probably want it to go to dev AND test, because the post deploy test should run against test